riscv/src/compiler.rs

```diff
@@ -544,11 +544,48 @@ fn preamble() -> String {
     // Removes up to 16 bits beyond 32
     // TODO is this really safe?
     instr wrap16 Y -> X { Y = Y_b5 * 2**32 + Y_b6 * 2**40 + X, X = X_b1 + X_b2 * 0x100 + X_b3 * 0x10000 + X_b4 * 0x1000000 }
-    constraints{
+    constraints {
         col witness Y_b5;
         col witness Y_b6;
+        col witness Y_b7;
+        col witness Y_b8;
         { Y_b5 } in { bytes };
         { Y_b6 } in { bytes };
+        { Y_b7 } in { bytes };
+        { Y_b8 } in { bytes };
+
+        col witness remainder; 
+
+        col witness REM_b1;
+        col witness REM_b2;
+        col witness REM_b3;
+        col witness REM_b4;
+        { REM_b1 } in { bytes };
+        { REM_b2 } in { bytes };
+        { REM_b3 } in { bytes };
+        { REM_b4 } in { bytes };
+    }
+
+    // implements Z = Y / X, stores remainder in `remainder`.
+    instr divu Y, X -> Z {
+        // Y is the known dividend
+        // X is the known divisor
+        // Z is the unknown quotient
+        // main division algorithm;
+        // if X is zero, remainder is set to dividend, as per RISC-V specification:
+        X * Z + remainder - Y = 0,
```

Comment:
```suggestion
        X * Z + remainder = Y;
```?

Reply:
How do you do these suggest changes?

Reply:
The +- icon

Reply:
If you click and drag over multiple lines, you can also suggest to change multiple lines at once.